src/main/java/org/opensearch/knn/plugin/KNNIndexSettingProvider.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.common.unit.ByteSizeUnit;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.index.shard.IndexSettingProvider;
+
+public class KNNIndexSettingProvider implements IndexSettingProvider {
+    public static final ByteSizeValue CHANGED_KNN_VALUE = new ByteSizeValue(16, ByteSizeUnit.MB);
+
+    @Override
+    public Settings getAdditionalIndexSettings(String indexName, boolean isDataStreamIndex, Settings templateAndRequestSettings) {
```

Comment:
This overridden method hooks into OpenSearch [here](https://github.com/opensearch-project/OpenSearch/blob/8ee5eebc3728a19487c0604d9828f92770fcd7ec/server/src/main/java/org/opensearch/node/Node.java#L1001).